openstf/pipeline/train_create_forecast_backtest.py

```diff
@@ -69,3 +205,38 @@ def train_model_and_forecast_back_test(
     forecast["horizon"] = test_data.iloc[:, -1]
 
     return forecast, model, train_data, validation_data, test_data
+
+
+def group_kfold(
+    input_data: pd.DataFrame, n_folds: int, random_split: bool = True
+) -> pd.DataFrame:
+    """Function to group data into groups, according to the date and the number of folds
+        - each date gets assigned a number between 0 and n_folds
+
+    Args:
+        input_data (pd.DataFrame): Input data
+        n_folds (int): Number of folds
+        random_split (bool): Indicates if random split needs to be applied
+
+    Returns:
+        grouped data (pandas.DataFrame)
+
+    """
+    unique_dates = input_data["dates"].unique()
+    # Group separators
+    len_data = len(unique_dates)
+    size = len_data // n_folds
+    rem = len_data % n_folds
```

Comment:
Can you add a bit more comments/documentation here? This will make it easier to see what is happening here